### PR TITLE
Hotfix/wy pubilc org jump back

### PR DESCRIPTION
**Pull request by sherotree**
## What type of PR is this?

- [ ] Feature
- [x] Bugfix
- [ ] Test
- [ ] Documentation
- [ ] Refactor
- [ ] Style
- [ ] Chore

## Which issue(s) this PR fixes:

Fixes #
Jump to the default organization if user is not in the current public organization

## Does this PR introduce a user interface change?

- [ ] Yes(screenshot is required)
- [ ] No

### CODE REVIEW

File: shell/app/org-home/services/org.tsx [hunk 1]
```diff
@@ -18,3 +18,10 @@ export const getOrgByDomain = (payload: ORG.IOrgReq): ORG.IOrg => {
     .query(payload)
     .then((response: any) => response.body);
 };
+
+export const getJoinedOrgs = (): IPagingResp<ORG.IOrg> => {
+  return agent
+    .get('/api/orgs')
+    .query({ pageNo: 1, pageSize: 30 })
```
Comment (daskyrk) [inline shell/app/org-home/services/org.tsx:25]: why 30?

File: shell/app/org-home/stores/org.tsx [hunk 2]
```diff
@@ -120,6 +122,10 @@ const org = createStore({
         }
       }
     },
+    async getJoinedOrgs({ call, update }) {
+      const { list } = await call(getJoinedOrgs);
+      update({ orgs: list });
+    },
```
Comment (daskyrk) [inline shell/app/org-home/stores/org.tsx:128]: This have already exist in userStore

File: shell/app/layout/pages/page-container/components/sidebar.tsx [hunk 3]
```diff
@@ -105,12 +105,12 @@ const AppCenterEl = () => {
...
 const SideBar = () => {
   const loginUser = userStore.useStore((s) => s.loginUser);
-  const currentOrg = orgStore.useStore(s => s.currentOrg);
+  const [currentOrg, orgs] = orgStore.useStore(s => [s.currentOrg, s.orgs]);
   const { switchMessageCenter } = layoutStore.reducers;
   const unreadCount = messageStore.useStore(s => s.unreadCount);
   // 清掉旧版本缓存
   window.localStorage.removeItem('dice-sider');
-  
+  const curOrgName = orgStore.useStore(s => s.currentOrg.name);
```
Comment (laojun) [inline shell/app/layout/pages/page-container/components/sidebar.tsx:113]: you've got currentOrg above, why get from orgStore again?

File: shell/app/layout/pages/page-container/components/sidebar.tsx [hunk 4]
```diff
@@ -149,6 +149,10 @@ const SideBar = () => {
     },
   ].filter(a => a.show);
 
+  useMount(() => {
+    orgStore.effects.getJoinedOrgs();
```
Comment (laojun) [inline shell/app/layout/pages/page-container/components/sidebar.tsx:153]: this effects only need to be executed once,  add a exist  condition to forbidden unnecessary request

File: shell/app/layout/pages/page-container/components/sidebar.tsx [hunk 5]
```diff
@@ -177,10 +181,15 @@ const SideBar = () => {
       layout="vertical"
       verticalBrandIcon={
         <img
-          className='mr0 pointer'
+          className="mr0 pointer"
           src={Logo}
           onClick={() => {
-            goTo(goTo.pages.orgRoot);
+            const isIncludeOrg = orgs.some((x: Obj) => x.name === curOrgName)
```
Comment (laojun) [inline shell/app/layout/pages/page-container/components/sidebar.tsx:187]: use find cloud be better

File: shell/app/layout/pages/page-container/components/sidebar.tsx [hunk 6]
```diff
@@ -177,10 +181,15 @@ const SideBar = () => {
...
+            if (isIncludeOrg) {
+              goTo(goTo.pages.orgRoot);
+            } else {
+              message.warning(i18n.t('default:org-jump-tip'), 2, () => goTo('/-'));
```
Comment (laojun) [inline shell/app/layout/pages/page-container/components/sidebar.tsx:191]: use Modal could be better?

Comment (laojun) [reply shell/app/layout/pages/page-container/components/sidebar.tsx:191]: and don't use '/-',  use goTo.pages.orgRoot

File: shell/app/org-home/stores/org.tsx [hunk 7]
```diff
@@ -120,6 +122,14 @@ const org = createStore({
         }
       }
     },
+    async getJoinedOrgs({ call, update }) {
+      const orgs = org.getState(s => s.orgs)
+      if (isEmpty(orgs)) {
+        return;
```
Comment (laojun) [inline shell/app/org-home/stores/org.tsx:128]: 1、condition error? !isEmpty(orgs) return; 
2、use select instead of org.getState;

**Review by laojun — APPROVED** [review]

**Pull request merged**